### DISCUSSION TITLE
Document - Biome integration as part of ESLint documentation page

### DISCUSSION THREAD

**Discussion by seedy** [vercel/next.js#59165] [Ideas]
### Goals

1. Become first-class citizen of Biome brand new toolkit based on Rust
2. Document how to integrate Biome as a replacement/alternative for Prettier
3. Document (in the near future) how to integrate Biome linter with NextJs as a possible alternative to ESLint

### Non-Goals

1. Change NextJs way of advocating core web vitals
2. Replace ESLint too early

### Background

Biome is a Rust based, ultra quick contender to Prettier. It won the de-facto legacy of Prettier by championing the recent Prettier bounty, launched by @vjeux himself.
It would be awesome to showcase this victory, knowing that Vercel teams are fond of Rust 😉

### Proposal

It should only be about documenting usage of Biome as a replacement or alternative to Prettier, as part of well-known code formatting tools.
Should be fairly easy to do based on the contributing guidelines, but I'm not sure I'll be quick enough to contribute. Feel free to do so in my stead, if you wish !

**Comment by shirakaba** [comment]
This is more than just a matter of documentation. As I detail in [this discussion](https://github.com/vercel/next.js/discussions/59347), `next lint` is coupled to eslint, so refactoring is necessary to adopt Biome without losing the other benefits of `next lint` (as it does more than just linting – it does doctoring, too).

**Reply by seedy** [reply]
While I agree with you, I think documenting biome integration is a quick and easy first step.
I handled the Biome setup alone for a NextJS project without considering `next lint`.

Therefore, I think it's possible to do both tasks independently.

I recommend keeping the door open for short iterations 😉

**Reply by shirakaba** [reply]
I took another approach of providing a no-op eslint implementation to satisfy `next lint` – this preserved the rest of `next lint`'s behaviour of sanity-checking your tsconfig, etc.

But indeed, that's totally valid, too – in fact, the Turbo 'basic' Next.js starter template uses `eslint` directly instead of `next lint`, which I suspect is an oversight by them 😅
